test/base_test.rb

```diff
@@ -1,50 +1,133 @@
+    assert_file_exists(btx_source_variables[:btx_log_path])
+    assert_command('ruby -I./lib ./bin/metababel -h')
   end
 
+  def subtest_generate_source_component
+    assert_command("ruby -I./lib ./bin/metababel -d #{btx_source_variables[:btx_model_path]} -t SOURCE -p #{btx_source_variables[:btx_pluggin_name]} -c #{btx_source_variables[:btx_component_name]} -o #{btx_source_variables[:btx_component_path]}")
+  end 
+
   def subtest_generate_source_callbacks
-    `ruby ./test/gen_source.rb -i #{btx_variables[:btx_target_log_path]} -o #{btx_variables[:btx_component_path]}/callbacks.c`
-    assert($?.success?)
+    assert_command("ruby ./test/gen_source.rb -i #{btx_source_variables[:btx_log_path]} -o #{btx_source_variables[:btx_component_path]}/callbacks.c")
   end
 
-  def subtest_compile_source
-    `cc -o #{btx_variables[:btx_component_path]}/#{btx_variables[:btx_pluggin_name]}_#{btx_variables[:btx_component_name]}.so #{btx_variables[:btx_component_path]}/*.c $(pkg-config --cflags babeltrace2) $(pkg-config --libs babeltrace2) -Wall -fpic --shared -I ./test/include/`
-    assert($?.success?)
+  def subtest_compile_source_component
+    assert_command("cc -o #{btx_source_variables[:btx_component_path]}/#{btx_source_variables[:btx_pluggin_name]}_#{btx_source_variables[:btx_component_name]}.so #{btx_source_variables[:btx_component_path]}/*.c $(pkg-config --cflags babeltrace2) $(pkg-config --libs babeltrace2) -Wall -fpic --shared -I ./test/include/")
   end
+  
+  def subtest_run_source_component
+    expected_output = File.open(btx_source_variables[:btx_log_path], 'r').read
+    assert_command_stdout(<<~TEXT, expected_output)
+      babeltrace2 --plugin-path=#{btx_source_variables[:btx_component_path]} \
+        --component=source.#{btx_source_variables[:btx_pluggin_name]}.#{btx_source_variables[:btx_component_name]}
+    TEXT
+  end
+end
 
-  def subtest_run_source
-    output = `babeltrace2 --plugin-path=#{btx_variables[:btx_component_path]} --component=#{btx_variables[:btx_component_type].downcase}.#{btx_variables[:btx_pluggin_name]}.#{btx_variables[:btx_component_name]}`
-    assert($?.success?)
+module SourceSubtestsDetail
+  # Generate a source with no messages.
+  def subtest_generate_source_callbacks
+    assert_command("ruby ./test/gen_source.rb -o #{btx_source_variables[:btx_component_path]}/callbacks.c")
+  end
 
-    expected_output = File.open(btx_variables[:btx_target_log_path], 'r').read
-    assert_equal(expected_output, output)
+  # Compare with text.details in place of log.
+  def subtest_run_source_component
+    expected_output = File.open(btx_source_variables[:btx_log_path], 'r').read
+    assert_command_stdout(<<~TEXT, expected_output)
+      babeltrace2 --plugin-path=#{btx_source_variables[:btx_component_path]} \
+        --component=source.#{btx_source_variables[:btx_pluggin_name]}.#{btx_source_variables[:btx_component_name]} \
+        --component=sink.text.details
+    TEXT
```

Comment:
what are those TEXT?

Reply:
What does `<<~` mean? I know `<<-` but not with the `~`. And why you hva a `,expecteout` after the text?

maybe do 
command=<<~TEST
      babeltrace2 --plugin-path=#{btx_source_variables[:btx_component_path]} \
        --component=source.#{btx_source_variables[:btx_pluggin_name]}.#{btx_source_variables[:btx_component_name]} \
        --component=sink.text.details
 TEST
 assert_command_stdout(command, expected_output);